.github/workflows/deploy.yaml

```diff
@@ -3,33 +3,88 @@ on:
     branches: [master, develop]
 
 jobs:
-  bump_version:
+  build:
     permissions:
       actions: "write"
-    name: "Bump pubspec version"
+    name: "Bump version and Build App Bundle"
     runs-on: ubuntu-latest
+    needs: [bump_version]
     env:
+      PROPERTIES_PATH: "android/key.properties"
+      JAVA_VERSION: "17.x"
+      FLUTTER_VERSION: "3.13.7"
       APP_VERSION_PATH: "uni/app_version.txt"
       PUBSPEC_PATH: "uni/pubspec.yaml"
+    defaults:
+      run:
+        working-directory: ./uni
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           token: ${{ secrets.NIAEFEUPBOT_PAT }}
+          fetch-depth: 0
 
       - name: Get develop hash
         if: github.ref == 'refs/heads/master'
         run: |
           git fetch origin develop
-          current_hash=$(git rev-parse origin/develop)
-          echo "DEVELOPHASH=$current_hash" >> $GITHUB_ENV
+          git pull origin master
+          echo "DEVELOP_HASH=$(git rev-parse origin/develop)" >> $GITHUB_ENV
+          echo "MASTER_HASH=$(git rev-parse origin/master^2)" >> $GITHUB_ENV 
```

Comment:
The last commit will always be a merge commit, we want the second parent (that comes from develop or another branch) instead of the first parent (which is the last commit from main)

Reply:
Ok, assuming that we will always use the merge strategy. Perhaps you can include a comment explaining just that